dumper/app/src/main/java/com/google/edwmigration/dumper/application/dumper/connector/oracle/OracleStatsConnector.java

```diff
@@ -30,19 +32,46 @@
 @ParametersAreNonnullByDefault
 public class OracleStatsConnector extends AbstractOracleConnector {
 
+  static final int MAX_DAYS = 10000;
+  static final Duration maxDuration = Duration.ofDays(MAX_DAYS);
+
   public OracleStatsConnector() {
     super(OracleConnectorScope.STATS);
   }
 
   @Override
   public void addTasksTo(List<? super Task<?>> out, ConnectorArguments arguments) throws Exception {
     StatsTaskListGenerator taskListGenerator = new StatsTaskListGenerator();
-    out.addAll(taskListGenerator.createTasks(arguments));
+    Duration queriedDuration = getQueriedDuration(arguments);
+    out.addAll(taskListGenerator.createTasks(arguments, queriedDuration));
   }
 
   @Nonnull
   @Override
   public String summary(String fileName) {
     return String.format("Oracle statistics saved to %s", fileName);
   }
+
+  static Duration getQueriedDuration(ConnectorArguments arguments) {
+    Duration queriedDuration = extractFromArgs(arguments);
+    if (queriedDuration.compareTo(Duration.ofDays(0)) == 1
```

Comment:
The expression `queriedDuration.compareTo(Duration.ofDays(0)) == 1` can be simplified to `DurationUtils.isPositive(queriedDuration)`.

Alternatively, both comparisons could be rewritten as:
```
Range.openClosed(ZERO, MAX_DURATION).contains(queriedDuration)
```

Reply:
The first suggestion has a different meaning. I'd like it to be "at least one day", rather than "positive duration".
In this case the difference doesn't change the behavior, because we operate on an integer number of days. However, the fact that these durations are always divisible by 1 day isn't clear to the reader and I don't see a good way to make it so.

The reason why the "at least one day" part is significant is that the only purpose of these Duration's is extracting the number of days. This means that if some future change led to a query with the duration being e.g. 6 hours, the actual query will use an empty time range and get no data.

Similarly, the `Range` can be used with `closed` instead of `openClosed` if we use a constant representing a 1-day duration. To me, neither option seems significantly better, please let me know if you have a preference.

Reply:
Correction: the original code with `queriedDuration.compareTo(Duration.ofDays(0)) == 1` is also wrong, it should be  `queriedDuration.compareTo(Duration.ofDays(1)) >= 0` and this discussion made me realize it. I'll update the code in a moment.

Reply:
> I'd like it to be "at least one day"
In this case, you could use `Range.closed(ofDays(1), MAX_DURATION)` as you already suggested in your comment.

> the fact that these durations are always divisible by 1 day isn't clear to the reader
I wouldn't worry about this aspect. The point is that the duration is converted to days, when being passed from Java to SQL query. The precision may be lost there, but I don't think that this would be surprising, if the original duration was not divisible by 1 day.

Reply:
Done